gr-digital/include/gnuradio/digital/clock_recovery_mm_cc.h

```diff
@@ -37,6 +37,18 @@ namespace gr {
      * This implements the Mueller and Müller (M&M) discrete-time
      * error-tracking synchronizer.
      *
+     * The input signal must have values around zero, as it works with
+     * sign changes of its input.
+     *
+     * Input can not be a square wave. Use blocks_moving_average_xx
+     * over samples per symbol to turn square wave into suitable input
+     * to this block.
+     *
```

Comment:
"The input signal must have peaks in order for the TED to output a correct error signal.  If the input signal pulses do not have peaks (e.g. rectangular pulses) the input signal should be conditioned with a matched pulse filter or other appropriate filter to peak the input pulses.  For a rectangular base pulse that is N samples wide, the matched filter taps would be [1.0/float(N)]*N"